### PR TITLE
Add `jobs_status` command.

### PR DESCRIPTION
**Pull request by lvwerra**
This command checks all logging dirs in a given path and lists all incomplete jobs (complete as well optionally) and their tasks status. Useful when running many `datatrove` pipelines in parallel to keep track which ones are complete or need relaunching.

**Example**:

```bash
/leandro/tokenize$ python status_pr.py /leandro/logs_dir/ --log_prefix tokenize_
           ❌tokenize_CC-MAIN-2014-23:                 63/64 completed tasks.                                                                                                                                                                               
           ❌tokenize_CC-MAIN-2014-23_merged:          0/1 completed tasks.                                                                                                                                                                                  
           ❌tokenize_CC-MAIN-2014-35:                 63/64 completed tasks.                                                                                                                                                                                
           ❌tokenize_CC-MAIN-2014-35_merged:          0/1 completed tasks.                                                                                                                                                                               
           ❌tokenize_CC-MAIN-2014-41:                 63/64 completed tasks.                                                                                                                                                                                
           ❌tokenize_CC-MAIN-2014-41_merged:          0/1 completed tasks.                                                                                                                                                                                  
           ❌tokenize_CC-MAIN-2014-42:                 63/64 completed tasks.                                                                                                                                                                                
           ❌tokenize_CC-MAIN-2014-42_merged:          0/1 completed tasks.                                                                                                                                                                                  
           Summary: 180/188 jobs completed.                                               
```

### CODE REVIEW

**Comment by guipenedo** [comment]
I like it! maybe add % as well, next to number of completed tasks/jobs?

File: src/datatrove/tools/jobs_status.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+parser = argparse.ArgumentParser("Fetch all jobs that are running or complete.")
+
+parser.add_argument(
+    "path", type=str, nargs="?", help="Path to the logging folder. Defaults to current directory.", default=os.getcwd()
+)
+
+parser.add_argument("--log_prefix", type=str, nargs="?", help="Prefix of logging folders to be scanned.", default="")
+parser.add_argument("--show_complete", help="Also list all jobs that are already complete.", action="store_true")
```
Comment (guipenedo) [inline src/datatrove/tools/jobs_status.py:33]: wouldn't it make sense to show them by default?

File: src/datatrove/tools/jobs_status.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        if not logging_dir.isfile("executor.json"):
+            console.log(
+                'Could not find "executor.json" in the given directory. Are you sure it is a ' "logging folder?",
+                style="red",
+            )
+            continue
+        with logging_dir.open("executor.json", "rt") as f:
+            world_size = json.load(f).get("world_size", None)
+        if not world_size:
+            console.log("Could not get the total number of tasks, please try relaunching the run.", style="red")
```
Comment (guipenedo) [inline src/datatrove/tools/jobs_status.py:63]: maybe also print the path so we know which job has a problem

File: src/datatrove/tools/jobs_status.py (hunk 1 above)
Comment (lvwerra) [reply src/datatrove/tools/jobs_status.py:32]: No strong opinion, in my case the complete ones were a lot so they polluted the list :)

File: src/datatrove/tools/jobs_status.py (hunk 2 above)
Comment (lvwerra) [reply src/datatrove/tools/jobs_status.py]: done!

File: src/datatrove/tools/jobs_status.py (hunk 1 above)
Comment (lvwerra) [reply src/datatrove/tools/jobs_status.py]: changed it!

**Comment by lvwerra** [comment]
Feedback should be addressed!

File: src/datatrove/tools/jobs_status.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ def list_folders_with_prefix(log_files_path, log_prefix):
 )
 
 parser.add_argument("--log_prefix", type=str, nargs="?", help="Prefix of logging folders to be scanned.", default="")
-parser.add_argument("--show_complete", help="Also list all jobs that are already complete.", action="store_true")
+parser.add_argument("--show_incomplete", help="Also list all jobs that are already complete.", action="store_true")
```
Comment (guipenedo) [inline src/datatrove/tools/jobs_status.py:32]: the `help=` probably needs updating

File: src/datatrove/tools/jobs_status.py [hunk 4]
```diff
@@ -51,14 +51,18 @@ def main():
         logging_dir = get_datafolder(path)
         if not logging_dir.isfile("executor.json"):
             console.log(
-                'Could not find "executor.json" in the given directory. Are you sure it is a ' "logging folder?",
+                f'Could not find "executor.json" in the given directory ({logging_dir}). Are you sure it is a '
```
Comment (guipenedo) [inline src/datatrove/tools/jobs_status.py:54]: `logging_dir` is a DataFolder object, maybe just `path`

File: src/datatrove/tools/jobs_status.py (hunk 1 above)
Comment (guipenedo) [reply src/datatrove/tools/jobs_status.py]: I think it's a bit confusing now, maybe just `hide_complete` with `store_false`?

File: src/datatrove/tools/jobs_status.py [hunk 5]
```diff
@@ -72,7 +76,7 @@ def main():
             emoji = "❌"
             incomplete_jobs += 1
 
-        if not (len(incomplete) == 0 and not args.show_complete):
+        if not (len(incomplete) == 0 and args.show_incomplete):
```
Comment (guipenedo) [inline src/datatrove/tools/jobs_status.py:79]: Check comment above about `hide_complete`
```suggestion
        if len(incomplete) > 0 or not args.hide_complete:
```

**Pull request merged**